File: sample/src/main/java/contacts/sample/cheatsheet/other/java/ContactLinksActivity.java
package contacts.sample.cheatsheet.other.java;

import android.app.Activity;

import java.util.List;

import contacts.core.ContactsFactory;
import contacts.core.aggregationexceptions.ContactLink;
import contacts.core.aggregationexceptions.ContactUnlink;
import contacts.core.entities.Contact;
import contacts.core.util.ContactLinksKt;

public class ContactLinksActivity extends Activity {

    ContactLink.Result link(List<Contact> contacts) {
        return ContactsFactory
                .create(this)
                .aggregationExceptions()
                .link()
                .contacts(contacts)
                .commit();
    }

    ContactUnlink.Result unlink(Contact contact) {
        return ContactsFactory
                .create(this)
                .aggregationExceptions()
                .unlink()
                .contact(contact)
                .commit();
    }

    ContactLink.Result linkDirect(List<Contact> contacts) {
        return ContactLinksKt.linkDirect(contacts, ContactsFactory.create(this));
    }

    ContactUnlink.Result unlinkDirect(Contact contact) {
        return ContactLinksKt.unlinkDirect(contact, ContactsFactory.create(this));
    }
}

File: sample/src/main/java/contacts/sample/cheatsheet/other/kotlin/ContactLinksActivity.kt
package contacts.sample.cheatsheet.other.kotlin

import android.app.Activity
import contacts.core.Contacts
import contacts.core.aggregationexceptions.ContactLink
import contacts.core.aggregationexceptions.ContactUnlink
import contacts.core.entities.Contact
import contacts.core.util.linkDirect
import contacts.core.util.unlinkDirect

class ContactLinksActivity : Activity() {

    fun link(contacts: List<Contact>): ContactLink.Result = Contacts(this)
        .aggregationExceptions()
        .link()
        .contacts(contacts)
        .commit()

    fun unlink(contact: Contact): ContactUnlink.Result = Contacts(this)
        .aggregationExceptions()
        .unlink()
        .contact(contact)
        .commit()

    fun linkDirect(contacts: List<Contact>): ContactLink.Result =
        contacts.linkDirect(Contacts(this))

    fun unlinkDirect(contact: Contact): ContactUnlink.Result =
        contact.unlinkDirect(Contacts(this))
}